snuba/datasets/metrics_messages.py

```diff
@@ -75,48 +77,68 @@ def value_for_counter_message(message: Mapping[str, Any]) -> Mapping[str, Any]:
 def aggregation_options_for_set_message(
     message: Mapping[str, Any], retention_days: int
 ) -> Mapping[str, Any]:
-    return {
-        "materialization_version": 1,
+    granularity_dict = {
         "granularities": [
             GRANULARITY_ONE_MINUTE,
             GRANULARITY_ONE_HOUR,
             GRANULARITY_ONE_DAY,
         ],
     }
 
+    if aggregation_setting := message.get("aggregation_option"):
+        parsed_aggregation_setting = AggregationOption(aggregation_setting)
+        if parsed_aggregation_setting is AggregationOption.TEN_SECOND:
+            granularity_dict["granularities"].append(GRANULARITY_TEN_SECONDS)
+            return {"materialization_version": 1, **granularity_dict}
+
+    return {"materialization_version": 1, **granularity_dict}
+
 
 def aggregation_options_for_distribution_message(
     message: Mapping[str, Any], retention_days: int
 ) -> Mapping[str, Any]:
-    aggregation_options = {
-        "min_retention_days": retention_days,
-        "materialization_version": 2,
+    granularity_dict = {
         "granularities": [
             GRANULARITY_ONE_MINUTE,
             GRANULARITY_ONE_HOUR,
             GRANULARITY_ONE_DAY,
         ],
     }
+    aggregation_options = {
+        "min_retention_days": retention_days,
+        "materialization_version": 2,
```

Comment:
Why is this different from sets ? 
Also has the new matview been applied everywhere ?

Reply:
Materialization version 2 was planned to be added for all metric types I believe (see https://github.com/getsentry/snuba/pull/4477/files) but it seems that never happened for sets and counters. (@john-z-yang maybe you have some information on this?)

The new materialization version would allow us to restrict the `retention_days` based on the granularity being stored. So this would address your above comment as well, once the matview is applied to all metric types. 

One option is to apply the new matviews to all the metric types now, or, we could do it as a follow-up from this PR. Do you have a preference for the order in which we do these?

Reply:
> Why is this different from sets ?
Also has the new matview been applied everywhere ?

It's only for distribution because we use v2 to disable histogram

Reply:
> (also, side question for John, do we know why the default values for the granularity-specific retention days columns, such as decasecond_retention_days, are different between distributions and sets/counters in this PR? For decasecond_retention_days I see 7 for distributions but retention_days for sets/counters. Am I missing something?)

It was a mistake, we were going to change it but since mat view v2 is not happening for the other 2, those values never did anything.

Reply:
> Materialization version 2 was planned to be added for all metric types I believe (see https://github.com/getsentry/snuba/pull/4477/files) but it seems that never happened for sets and counters.
Yeah it was supposed to get applied on the other metric types as well. But priorities changed and we weren't able to apply it everywhere. We should fix this since now that there is a need to control configurations across all datasets.

Reply:
Yeah, I agree. Do we have preferences on how this work should be done? Could this PR go in first and open up 10s granularity for a bit (and also advise the relevant product team to send in a lower `retention_days` in their metrics)? Or do we think these re-configurations should happen first?

Reply:
> Yeah, I agree. Do we have preferences on how this work should be done? Could this PR go in first and open up 10s granularity for a bit (and also advise the relevant product team to send in a lower retention_days in their metrics)? Or do we think these new materialized views should go in first?

Those 2 things can be decoupled from each other. There isn't a need to sequence them.